saltswap/tests/test_simulation.py

```diff
@@ -1,11 +1,191 @@
+
+        return integrator, context, salinator
+
+    def test_protein_initialization(self):
+        """
+        Test the initialization of the salinator
+        """
+        integrator, context, salinator = self._create_protein_system()
+
+    def test_waterbox_initialization(self):
+        """
+        Implicitly tests the identification of tip3p and tip4p water models, as exceptions are thrown if the water
+        model is not recognized during the class initialization.
+        """
+        integrator, context, salinator = self._create_waterbox(model='tip4pew')
+        integrator, context, salinator = self._create_waterbox(model='tip3p')
+
+    def test_invert_concentration(self):
+        """
+        Make sure that when calculating the chemical potential from the concentration, the step that involves inverting
+        the chemical potential-->concentration is correctly performed.
+        """
+        integrator, context, salinator = self._create_waterbox(model='tip4pew')
+
+        # Get the default weights for tip4pew water
+        weights = wrappers.default_tip4pew_weights
+        fn = weights['fn']
+        volume = weights['volume']
+
+        # Get the concentration for a specified chemical potential
+        delta_chem = 315.0
+        predicted_concentration = salinator.predict_concentration(delta_chem, fn, volume)
+
+        # Get the chemical potential from the concentration
+        predicted_delta_chem = salinator.invert_concentration(predicted_concentration, fn, volume)
+
+        assert abs(predicted_delta_chem - delta_chem) < 1E-6
+
+    def test_charge_calculation(self):
+        """
+        Verify type of total charge calculation.
+        """
+        integrator, context, salinator = self._create_protein_system()
+
+        nonbonded_force = salinator._get_nonbonded_force()
+        initial_charge = salinator._get_system_charge(nonbonded_force)
+
+        # There should be a non-zero total charge for the DFHR test system
+        assert type(initial_charge) == int and initial_charge != 0
+
+    def test_neutralizer(self):
+        """
+        Make sure the system can be correctly neutralized.
+        """
+        integrator, context, salinator = self._create_protein_system()
+
+        # Neutralize
+        salinator.neutralize()
+
+        # Get the new charge
+        nonbonded_force = salinator._get_nonbonded_force()
+        final_charge = salinator._get_system_charge(nonbonded_force)
+
+        assert final_charge == 0
+
+    def test_initial_insertion(self):
+        """
+        Test to make sure salt gets added when an initial guess of the number is made.
+        """
+        integrator, context, salinator = self._create_waterbox()
+
+        # Neutralize
+        salinator.neutralize()
+
+        # Get the initial number of salt
+        initial_nsalt = np.sum(salinator.swapper.stateVector == 1)
+
+        # Add salt to the specified concentration
+        salinator.initialize_concentration()
```

Comment:
The salt concentration is a property of the `Salinator` object, and it must be explicitly stated when initializing the class. However, I'll still have a think whether this is worth adding as an argument here.